pkg/controller/backup-operator/s3_backup.go

```diff
@@ -37,23 +35,12 @@ func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, namespace, cluster
 		return "", err
 	}
 	defer cli.Close()
-	prefix := backupapi.ToS3Prefix(s3.Prefix, namespace, clusterName)
-	s3cli := backups3.NewFromClient(s3.S3Bucket, prefix, cli.S3)
-	be := backend.NewS3Backend(s3cli)
-	bm := backup.NewBackupManager(kubecli, clusterName, namespace, nil, be)
-
-	// this SaveSnap takes 0 as lastSnapRev to indicate that it
-	// saves any snapshot with revision > 0.
-	bs, err := bm.SaveSnap(0)
+	// TODO: support TLS.
+	bm := backup.NewBackupManagerFromWriter(kubecli, nil, writer.NewS3Writer(cli.S3), clusterName, namespace)
+	s3Prefix := backupapi.ToS3Prefix(s3.Prefix, namespace, clusterName)
+	_, fullPath, err := bm.SaveSnapWithPrefix(path.Join(s3.S3Bucket, s3Prefix))
```

Comment:
No need to return Size?

Reply:
size is not used anywhere. so no need to return it.